SYCL/AtomicRef/add.cpp

```diff
@@ -1,44 +1,154 @@
-// RUN: %clangxx -fsycl -fsycl-unnamed-lambda -fsycl-targets=%sycl_triple %s -o %t.out \
-// RUN: -Xsycl-target-backend=nvptx64-nvidia-cuda --cuda-gpu-arch=sm_60
+// Each combination of 64/32 bit atomic, relaxed/acquire/release/acq_rel
+// semantic order and sub_group/work_group/device/system scope is tested
+// separately. This is controlled by macros, defined by RUN commands. Defaults
+// (no macro for a group) are: 32 bit, relaxed and device.
+
+// RUN: %clangxx -fsycl -fsycl-unnamed-lambda -fsycl-targets=%sycl_triple %s -o %t.out -Xsycl-target-backend --cuda-gpu-arch=sm_60
```

Comment:
```
$ "./llvm.obj/bin/clang++" "-fsycl" "-fsycl-unnamed-lambda" "-fsycl-targets=spir64" "./llvm-test-suite/SYCL/AtomicRef/xor_generic.cpp" "-o" "./build/SYCL/AtomicRef/Output/xor_generic.cpp.tmp.out" "-Xsycl-target-backend" "--cuda-gpu-arch=sm_60"
# command stderr:
clang++: /netbatch/donb3330453_00/runDir/jenkins-dir/workspace/Tools_SH/SYCLOS_Nightly/RHEL/llvm.src/clang/lib/Driver/Driver.cpp:7200: clang::driver::InputInfo clang::driver::Driver::BuildJobsForActionNoCache(clang::driver::Compilation&, const clang::driver::Action*, const clang::driver::ToolChain*, llvm::StringRef, bool, bool, const char*, std::map<std::pair<const clang::driver::Action*, std::__cxx11::basic_string<char> >, clang::driver::InputInfo>&, clang::driver::Action::OffloadKind) const: Assertion `CachedResults.find(ActionTC) != CachedResults.end() && "Result does not exist??"' failed.
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:
0.	Program arguments: ./llvm.obj/bin/clang++ -fsycl -fsycl-unnamed-lambda -fsycl-targets=spir64 ./llvm-test-suite/SYCL/AtomicRef/xor_generic.cpp -o ./build/SYCL/AtomicRef/Output/xor_generic.cpp.tmp.out -Xsycl-target-backend --cuda-gpu-arch=sm_60
1.	Compilation construction
2.	Building compilation jobs
3.	Building compilation jobs
4.	Building compilation jobs
5.	Building compilation jobs
6.	Building compilation jobs
7.	Building compilation jobs
8.	Building compilation jobs
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
./llvm.obj/bin/clang++[0x20716cf]
./llvm.obj/bin/clang++[0x206ef69]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x153c0)[0x1525036343c0]
/lib/x86_64-linux-gnu/libc.so.6(gsignal+0xcb)[0x1525030ca18b]
/lib/x86_64-linux-gnu/libc.so.6(abort+0x12b)[0x1525030a9859]
/lib/x86_64-linux-gnu/libc.so.6(+0x25729)[0x1525030a9729]
/lib/x86_64-linux-gnu/libc.so.6(+0x36f36)[0x1525030baf36]
```

I suppose the problem here is `-Xsycl-target-backend --cuda-gpu-arch=sm_60`. 
-fsycl-targets=%sycl_triple depends on the device we run the test on. LIT uses  SPIR triple for running the test on Intel devices and it's probably not compatible with target back-end option.

If this test is supposed to be run on NVIDIA gpu only, I suggest adding `REQUIRES: cuda`.

Reply:
> If this test is supposed to be run on NVIDIA gpu only, I suggest adding REQUIRES: cuda

For now only cuda supports all the combinations of memory orders and scopes so that should be fine. I imagine eventually other backends will support that as well.

Reply:
I looked at how these tests are executed now and it looks like they cover all platforms and devices. I think we should not reduce the existing scope of testing.
Please, keep running supported "combinations of memory orders and scopes" on all devices.
To fix the compilation, you should specify the backend target for CUDA specific options: `-Xsycl-target-backend=nvptx64-nvidia-cuda --cuda-gpu-arch=sm_60`.

Reply:
That seems like a better option, thanks.